### ISSUE TITLE
SST DEV results in :443 connection refused (2)

### ISSUE THREAD

**Issue by cosmic-ascendant** [anomalyco/sst#5187]
Having issues with lambda invocations. Running `sst/examples/aws-lambda-stream` out the box. Resources are created without error, but upon invoking, lambda logs give:

```
2024/12/08 01:18:28 INFO waiting for interrupt signal
2024/12/08 01:18:28 ERROR run failed err="dial tcp :443: connect: connection refused"
INIT_REPORT Init Duration: 99.38 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
```

This was closed in https://github.com/sst/sst/issues/5184

sst version: 3.3.59

**Comment by cosmic-ascendant** [comment]
I noticed these two env vars were unset in the lambda:
```
SST_APPSYNC_HTTP
SST_APPSYNC_REALTIME
```
while I do see them being resolved in sst.log
```
time=2024-12-08T09:00:09.947-10:00 level=INFO msg="found appsync" rest=<id>.appsync-api.us-west-2.amazonaws.com realtime=<id>.appsync-realtime-api.us-west-2.amazonaws.com"
```
I've also tried "rebooting" the boostrap components by removing assets/state in ECR, S3, AppSync, and Parameter Store.

Open to any suggestions for further debugging steps.

**Comment by jyungtong** [comment]
downgrading to 3.2.76 works again

**Comment by hbswift** [comment]
> downgrading to 3.2.76 works again

Somehow still have the same issue on that version. Could be me doing something wrong.

**Comment by cosmic-ascendant** [comment]
> downgrading to 3.2.76 works again

I've tried downgrading to other 3.2.X versions and still facing same issue.

**Comment by ngocmanh1712** [comment]
any update here? I am also facing this too.

**Comment by gel1123** [comment]
I encountered the same issue with SST version 3.3.61. However, in my case, downgrading to version 3.2.76 resolved the problem.

**Comment by hbswift** [comment]
[v3.3.64](https://github.com/sst/sst/releases/tag/v3.3.64) tries to resolve this issue but I still have the same outcome.

**Comment by theoneamendez** [comment]
v3.3.64 does not resolve the issue for me either

Edit: actually did work. I was using the example aws-python in examples folder and that wasn't working out of the box because of the dependencies couldn't load. Cleared all dependencies from src/pyproject.toml and was able to run, probably errored because of issue with sst dependency.

**Comment by jyungtong** [comment]
3.3.65 work for me

**Comment by ngocmanh1712** [comment]
> 3.3.65 work for me

I updated 3.3.65 but still got this error. Tried deleting dev stage, node_module, sst too but nothing change
```
{"body":"sst dev is not running"}
```

**Comment by hbswift** [comment]
@ngocmanh1712 - I had the same issue but it was just a matter of making a change in my lambda and having it rebuild itself. All works now!

**Comment by ngocmanh1712** [comment]
> @ngocmanh1712 - I had the same issue but it was just a matter of making a change in my lambda and having it rebuild itself. All works now!

I am still getting error. Also have matter of auto rebuilding lambda code, it doesn't work for me.
What version did you use which fixed all of errors?

**Comment by jayair** [comment]
@ngocmanh1712 or anybody else still having an issue with the latest sst?

**Comment by cosmic-ascendant** [comment]
@jayair Ok, just tried SST v3.4.3. I had previously tried a few 3.2.x and 3.3.x build without success.

 Keeping it simple, trying a fresh /examples/aws-lambda-stream -> npm install -> npx sst dev.

I noticed the initial deployment took a good bit longer (deleting and starting worker from logs). Deployed fine, hit the url. Got a couple "sst dev is not running" responses. Looked at lambda deployment and noted this time the env vars `SST_APPSYNC_HTTP` and `SST_APPSYNC_REALTIME` are actually populated now. Saw 200's in lambda logs so invoked the url again and got the expected response.

So, tldr, seems good! Probably fixed from some of the "dev: improve sst dev not connected scnearios" patches. Thanks.

**Comment by ngocmanh1712** [comment]
> @ngocmanh1712 or anybody else still having an issue with the latest sst?

I still have having "sst dev is not running" issue with latest 3.4.16

Lambda logs
![image](https://github.com/user-attachments/assets/4cf7cc37-02e3-4010-985e-2fd07249c44c)

when call api, it returns this
![image](https://github.com/user-attachments/assets/2e81ca22-24af-4eaf-83ae-7abbc3fc1ea8)

**Comment by jayair** [comment]
Hmm we might need some logs for this. @thdxr

**Comment by thdxr** [comment]
this should be fixed
